### PR TITLE
gh-90822: Redefined `PY_SSIZE_T_MAX` and `PY_SSIZE_T_MIN`

### LINKED ISSUE #90822: PY_SSIZE_T_MAX is not an integer constant expression
BPO | [46664](https://bugs.python.org/issue46664)
--- | :---

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2022-02-06.21:20:42.705>
labels = ['3.7', '3.8', '3.9', '3.10', '3.11', 'expert-C-API', 'type-feature']
title = 'PY_SSIZE_T_MAX is not an integer constant expression'
updated_at = <Date 2022-02-06.21:37:17.876>
user = 'https://bugs.python.org/ov2k'
```

bugs.python.org fields:
```python
activity = <Date 2022-02-06.21:37:17.876>
actor = 'ov2k'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['C API']
creation = <Date 2022-02-06.21:20:42.705>
creator = 'ov2k'
dependencies = []
files = []
hgrepos = []
issue_num = 46664
keywords = []
message_count = 1.0
messages = ['412670']
nosy_count = 1.0
nosy_names = ['ov2k']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue46664'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

### PR DESCRIPTION
**Pull request by ganesh-k13**
Fixes: gh-90822

### CODE REVIEW

**Comment by ganesh-k13** [comment]
Hey, @methane (apologies for direct ping), I'm new here, I assume `skip news` would mean I can remove the news file?

**Comment by methane** [comment]
Yes. You can remove the news file.

File: Include/pyport.h [hunk 1]
```diff
@@ -114,17 +114,23 @@ typedef intptr_t        Py_intptr_t;
...
+#   define PY_SSIZE_T_MAX SSIZE_MAX
 #elif SIZEOF_VOID_P == SIZEOF_SIZE_T
 typedef Py_intptr_t     Py_ssize_t;
+#   define PY_SSIZE_T_MAX INTPTR_MAX
 #else
 #   error "Python needs a typedef for Py_ssize_t in pyport.h."
 #endif
 
+/* Smallest negative value of type Py_ssize_t. */
+#define PY_SSIZE_T_MIN (-PY_SSIZE_T_MAX-1)
```
Comment (methane) [inline Include/pyport.h:132]: How about changing MIN too?

```
# define PY_SSIZE_T_MIN SSIZE_MIN
...
# define PY_SSIZE_T_MIN INTPTR_MIN
```

Comment (methane) [reply Include/pyport.h:132]: Uh, it is cleaner but needs 4 definition, not 2.

Comment (ganesh-k13) [reply Include/pyport.h:132]: There were two reasons:
1. My system did not have a `SSIZE_MIN`, maybe some issue?
2. We would have to make a `#define` in the windows portion as well, which ok TBH, but we can make do with just 1.

I'm ok with both approaches, what do you think?

**Review by methane — APPROVED** [review]
LGTM

**Pull request merged**